tests/test_repository.py

```diff
@@ -1128,6 +1128,30 @@ def test_auto_track_large_files(self):
             is_tracked_with_lfs(os.path.join(WORKING_REPO_DIR, "small_file.txt"))
         )
 
+    def test_auto_track_binary_files(self):
+        repo = Repository(WORKING_REPO_DIR)
+
+        # This content is non-binary
+        non_binary_file = [100] * int(1e6)
+
+        # This content is binary (contains the null character)
+        binary_file = "\x00\x00\x00\x00"
+
+        with open(f"{WORKING_REPO_DIR}/non_binary_file.txt", "w+") as f:
```

Comment:
we shouldn't write on the working dir, we should work with python's temp files and folders.

Reply:
The `WORKING_REPO_DIR` is a folder that is created on the start of the test, and which is reset at every beginning of every test, see here: https://github.com/huggingface/huggingface_hub/blob/6d27a15cb3b8f9d6b6b87075701192569055136f/tests/test_repository.py#L76-L91

This isn't the working directory in which the test is run, but a folder nested in the `fixtures` folder:

https://github.com/huggingface/huggingface_hub/blob/6d27a15cb3b8f9d6b6b87075701192569055136f/tests/test_repository.py#L50-L52